tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,327 @@
+"""
+Test the auto-restart feature of containers 
+"""
+import time
+import pytest
+import logging
+
+from collections import defaultdict
+
+from common.utilities import wait_until
+from common.helpers.assertions import pytest_assert
+
+logger = logging.getLogger(__name__)
+
+CONTAINER_CHECK_INTERVAL_SECS = 1
+CONTAINER_STOP_THRESHOLD_SECS = 30
+CONTAINER_RESTART_THRESHOLD_SECS = 180
+
+CMD_CONTAINER_FEATURE_AUTORESTART = "show container feature autorestart"
+
+def get_critical_group_and_process_list(duthost, container_name):
+    """
+    @summary: Get critical group and process lists by parsing the 
+              critical_processes file in the specified container
+    @return: Two lists which include the critical process and critical groups respectively
+    """
+    critical_group_list = []
+    critical_process_list = []
+
+    file_content = duthost.shell("docker exec {} bash -c '[ -f /etc/supervisor/critical_processes ] \
+        && cat /etc/supervisor/critical_processes'".format(container_name), module_ignore_errors=True)
```

Comment:
Suggest to check for the existence of the file using `duthost.stat()`

Reply:
Scratch that. Since the critical_processes file resides in a container, `duthost.stat()` will not work, unfortunately.

Reply:
Let me think for a solution.